cookbook/session/locale_sticky_session.rst

```diff
@@ -106,3 +106,93 @@ method::
     {
         $locale = $request->getLocale();
     }
+
+Setting the Locale based on the User Entity
+-------------------------------------------
+
+You might want to improve this technique even further and define the locale based on
+the user entity of the logged in user. However since the ``LocaleListener`` is called
+before the ``FirewallListener``, which is responsible for handling authentication and
+is setting the user token into the ``TokenStorage``, you have no access to the user
+which is logged in.
+
+First lets pretend you have defined a property locale in your user entity which you
+want to be used as the locale for the given user. In order to achieve the wanted locale
+configuration you can set the locale which is defined for the user to the session right
+after the login. Fortunately you can hook into the login process and update your session
+variable before the redirect to the first page. For this you need an event listener for the
```

Comment:
I would remove "variable" here, but right something like "[...] and update the user's session before [...]"